cypress-tests/cypress/support/commands.js

```diff
@@ -258,10 +260,191 @@ Cypress.Commands.add("confirmCallTest", (confirmBody, req_data, res_data, confir
         expect(res_data.body.error[key]).to.equal(response.body.error[key]);
       }
     }
-    
   });
 });
 
+Cypress.Commands.add(
+  "confirmBankRedirectCallTest",
+  (confirmBody, req_data, res_data, confirm, globalState) => {
+    const paymentIntentId = globalState.get("paymentID");
+    confirmBody.payment_method = req_data.payment_method;
+    confirmBody.payment_method_type = req_data.payment_method_type;
+    confirmBody.payment_method_data.bank_redirect = req_data.bank_redirect;
+    confirmBody.confirm = confirm;
+    confirmBody.client_secret = globalState.get("clientSecret");
+
+    cy.request({
+      method: "POST",
+      url: `${globalState.get("baseUrl")}/payments/${paymentIntentId}/confirm`,
+      headers: {
+        "Content-Type": "application/json",
+        "api-key": globalState.get("publishableKey"),
+      },
+      failOnStatusCode: false,
+      body: confirmBody,
+    }).then((response) => {
+      logRequestId(response.headers["x-request-id"]);
+      expect(res_data.status).to.equal(response.status);
+      expect(response.headers["content-type"]).to.include("application/json");
+      globalState.set("paymentID", paymentIntentId);
+
+      switch (response.body.authentication_type) {
+        case "three_ds":
+          if (
+            response.body.capture_method === "automatic" ||
+            response.body.capture_method === "manual"
+          ) {
+            if (response.body.status !== "failed") { // we get many statuses here, hence this verification
+              if (response.body.payment_method_type === "blik") {
+                expect(response.body)
+                  .to.have.property("next_action")
+                  .to.have.property("type")
+                  .to.equal("wait_screen_information");
+              } else {
+                expect(response.body)
+                  .to.have.property("next_action")
+                  .to.have.property("redirect_to_url");
+                globalState.set(
+                  "nextActionUrl",
+                  response.body.next_action.redirect_to_url
+                );
+              }
+            } else if (response.body.status === "failed") {
+              expect(response.body.error_code).to.equal(res_data.body.error_code);
+            }
+          } else {
+            throw new Error(
+              `Unsupported capture method; ${response.body.capture_method}`
+            );
```

Comment:
instead of throwing error, have it expect error otherwise the flow will break for error cases
```suggestion
            expect(response.body).to.have.property("error");
            for(const key in res_data.body.error) {
              expect(res_data.body.error[key]).to.equal(response.body.error[key]);
            }
```